src/core/qgis.h

```diff
@@ -330,6 +330,41 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( VectorExportResult )
 
+    /**
+     * SqlLayerDefinitionCapability enum lists the arguments supported by the provider when creating SQL query layers.
+     * \since QGIS 3.22
+     */
+    enum class SqlLayerDefinitionCapability : int
+    {
+      Filter = 1 << 1,            //!< SQL layer definition supports filter
+      GeometryColumn = 1 << 2,    //!< SQL layer definition supports geometry column
+      PrimaryKeys = 1 << 3,       //!< SQL layer definition supports primary keys
+      SelectAtId = 1 << 4         //!< SQL layer definition supports disabling select at id
```

Comment:
Again, I'd love for us to use different wording here and avoid propagating the unclear term through more of the API

Reply:
Is that actually used by any provider? PG apparently doesn't use it.